File: conformance/bundles/09-expected-message-mismatch/boom.steps.ts
import { steps } from '@varar/varar'

const { stimulus } = steps(() => ({}))

// Throws a message that does NOT contain the expected substring "expected
// message", so the expected-failure is NOT satisfied → the example fails.
stimulus('I always boom', () => {
  throw new Error('actual different error')
})

File: conformance/bundles/09-expected-message-mismatch/boom.steps.rs
//! Rust sibling of `boom.steps.ts` (bundle `09-expected-message-mismatch`).

use varar::{HandlerError, Steps, Value};

pub fn register(s: &mut Steps) {
    // Throws a message that does NOT contain the expected substring "expected
    // message", so the expected-failure is NOT satisfied → the example fails.
    s.stimulus("I always boom", |_state| Err(HandlerError::new("actual different error")));
}

pub fn state() -> Value {
    Value::Null
}
